src/besselk.jl

```diff
@@ -431,3 +434,68 @@ end
 besselk_power_series_cutoff(nu, x::Float64) = x < 2.0 || nu > 1.6x - 1.0
 besselk_power_series_cutoff(nu, x::Float32) = x < 10.0f0 || nu > 1.65f0*x - 8.0f0
 
+
+"""
+  besselk_asymptoticexp(v, x, order)
+
+Computes the asymptotic expansion of K_ν w.r.t. argument. Accurate for large x, and faster than uniform asymptotic expansion for small to small-ish orders. The default order of the expansion is 10.
+"""
+function besselk_asymptoticexp(v, x::T) where T 
+  _besselk_asymptoticexp(v, float(x), 10)
```

Comment:
Is this extra function needed? x should already be a float when we call this function so seems like just extra confusion. I think we include the order `10` in the top level call so it's more clear.

Reply:
No, probably not. Honestly I don't really understand the purpose of the underscore and then `float(x)` things in the other functions so I was sort of just robotically doing that. I'll cut it out in the next commit.

Reply:
We just want the function to take inputs like `f(4), f(4.0)` etc. Since we usually only supply implementations for `Float32` or `Float64` it just provides a nice way for the top level function to take a lot of different valid types. But with these internal functions don't really need that.

Reply:
Ah, that makes sense. Taken care of in this newest commit.